src/topology/alexandroff.lean

```diff
@@ -0,0 +1,370 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang, Yury Kudryashov
+-/
+import topology.separation
+import topology.opens
+
+/-!
+# The Alexandroff Compactification
+
+We construct the Alexandroff compactification of an arbitrary topological space `X` and prove
+some properties inherited from `X`.
+
+## Main defintion
+
+* `alexandroff`: the Alexandroff compactification, we use coercion for the canonical embedding
+  `X → alexandroff X`;
+* `alexandroff.infty`: the extra point
+
+## Main results
+* The topological structure of `alexandroff X`
+* The connectedness of `alexandroff X` for a noncompact, preconnected `X`
+* `alexandroff X` is `T₀` for a T₀ space `X`
+* `alexandroff X` is `T₁` for a T₁ space `X`
+* `alexandroff X` is normal if `X` is locally compact and Hausdorff
+
+## Tags
+
+one-point compactification, compactness
+-/
+
+open set filter
+open_locale classical topological_space filter
+
+/-!
+### Definition and basic properties
+
+In this section we define `alexandroff X` to be the disjoint union of `X` and `∞`, implemented as
+`option X`. Then we restate some lemmas about `option X` for `alexandroff X`.
+-/
+
+/-- The Alexandroff extension of an arbitrary topological space `X` -/
+def alexandroff (X : Type*) := option X
+
+namespace alexandroff
+
+variables {X : Type*}
+
+/-- The point at infinity -/
+def infty : alexandroff X := none
+localized "notation `∞` := alexandroff.infty" in alexandroff
+
+instance : has_coe_t X (alexandroff X) := ⟨option.some⟩
+
+instance : inhabited (alexandroff X) := ⟨∞⟩
+
+lemma coe_injective : function.injective (coe : X → alexandroff X) :=
+option.some_injective X
+
+@[simp, norm_cast] lemma coe_eq_coe {x y : X} : (x : alexandroff X) = y ↔ x = y :=
+coe_injective.eq_iff
+
+@[simp] lemma coe_ne_infty (x : X) : (x : alexandroff X) ≠ ∞  .
+@[simp] lemma infty_ne_coe (x : X) : ∞ ≠ (x : alexandroff X) .
+@[simp] lemma some_eq_coe (x : X) : some x = (x : alexandroff X) := rfl
```

Comment:
This seems to be an overly far-reaching simp lemma.  For example, it will apply here:
```lean
example (x : ℕ) (z : option ℕ) : some x = z :=
begin
  simp,
  -- now it's (x : alexandroff ℕ) = z
end
```
The next lemma (`none_eq_infty`) has the same issue.